sim/common/caster_trinkets.go

```diff
@@ -7,138 +7,235 @@ import (
-	core.AddActiveItem(29132, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDSpellPower, 150, time.Second*15), ActivateCD: time.Second * 90, CoolID: core.MagicIDScryerTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(29179, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDSpellPower, 150, time.Second*15), ActivateCD: time.Second * 90, CoolID: core.MagicIDXiriTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(29370, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDBlessingSilverCrescent, 155, time.Second*20), ActivateCD: time.Second * 120, CoolID: core.MagicIDISCTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(29376, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDSpellPower, 99, time.Second*20), ActivateCD: time.Second * 120, CoolID: core.MagicIDEssMartyrTrink, SharedID: core.MagicIDHealTrinket})
-	core.AddActiveItem(30626, core.ActiveItem{BuffUp: ActivateSextant, ActivateCD: core.NeverExpires, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(31856, core.ActiveItem{BuffUp: ActivateDCC, ActivateCD: core.NeverExpires, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(32483, core.ActiveItem{Activate: core.CreateHasteActivate(core.MagicIDSkullGuldan, 175, time.Second*20), ActivateCD: time.Second * 120, CoolID: core.MagicIDSkullGuldanTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(33829, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDHexShunkHead, 211, time.Second*20), ActivateCD: time.Second * 120, CoolID: core.MagicIDHexTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(34429, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDShiftingNaaru, 320, time.Second*15), ActivateCD: time.Second * 90, CoolID: core.MagicIDShiftingNaaruTrink, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(35749, core.ActiveItem{BuffUp: ActivateAlchStone, ActivateCD: core.NeverExpires, SharedID: core.MagicIDAtkTrinket})
-	core.AddActiveItem(38290, core.ActiveItem{Activate: core.CreateSpellDmgActivate(core.MagicIDDarkIronPipeweed, 155, time.Second*20), ActivateCD: time.Second * 120, CoolID: core.MagicIDDITrink, SharedID: core.MagicIDAtkTrinket})
+	// Proc effects. Keep these in order by item ID.
+	core.AddItemEffect(23207, ApplyMarkOfTheChampion)
+	core.AddItemEffect(27683, ApplyQuagmirransEye)
+	core.AddItemEffect(28418, ApplyShiffarsNexusHorn)
+	core.AddItemEffect(28789, ApplyEyeOfMagtheridon)
+	core.AddItemEffect(30626, ApplySextantOfUnstableCurrents)
+	core.AddItemEffect(31856, ApplyDarkmoonCardCrusade)
+
+	// Activatable effects. Keep these in order by item ID.
+	core.AddItemEffect(23046, core.MakeTemporaryStatsOnUseCDRegistration(
+		core.MagicIDSpellPower,
+		stats.SpellPower,
+		130,
+		time.Second*20,
+		core.MajorCooldown{
+			CooldownID: core.MagicIDEssSappTrink,
+			Cooldown: time.Minute * 2,
+			SharedCooldownID: core.MagicIDAtkTrinket,
+		},
+	))
```

Comment:
this change to the item effects seems slightly less readable (since its now more a function call than a struct with key names)... but its not something that has to be reviewed or changed often so it shouldn't be a big deal.